community-tutorials/ruby-on-rails-in-production/01-en.md

```diff
@@ -0,0 +1,269 @@
+
+
+The easiest way is to clone rbenv from Github, so we have the latest version. To do this, we proceed as follows:
+
+```sh
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+echo 'export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"' >> ~/.bashrc
+git clone https://github.com/rbenv/rbenv-vars.git ~/.rbenv/plugins/rbenv-vars
+exec $SHELL
+```
+
+Once rbenv is installed we can continue with the installation of the Ruby version, in our example we install the current version 3.0.2, but this can also be changed to any version. 
+
+```sh
+rbenv install 3.0.2
+rbenv global 3.0.2
+ruby -v
+```
+
+If the version 3.0.2 appears after executing `ruby -v`, this has worked.
+
+## Installing Bundler
+
+In order to be able to install gems from the project in the future, we also need to install the bundler.
+
+```sh
+gem install bundler
+bundle -v
+```
+
+If the Bundler version 2 is now displayed again, this has worked and we have completed the installation of ruby.
+
+
+# Configuring Web Server
+
+
+For our production environment we use the NGINX to receive http requests. The http requests are then passed on to Passenger, which ensures that this is executed in our Rails application.
+
+## Installing NGINX & Passenger
+
+To do this, we first have to add repositories to the system again:
+
+```sh
+sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 561F9B9CAC40B2F7
+sudo sh -c 'echo deb https://oss-binaries.phusionpassenger.com/apt/passenger focal main > /etc/apt/sources.list.d/passenger.list'
+```
+
+Then we update the list again:
+
+```sh
+sudo apt-get update
+```
+
+Now we can install NGINX and Passenger:
+
+```sh
+sudo apt-get install -y nginx-extras libnginx-mod-http-passenger
+```
+
+To activate Passenger in NGINX we have to execute the following 2 commands:
+
+```sh
+if [ ! -f /etc/nginx/modules-enabled/50-mod-http-passenger.conf ]; then sudo ln -s /usr/share/nginx/modules-available/mod-http-passenger.load /etc/nginx/modules-enabled/50-mod-http-passenger.conf ; fi
+```
+
+```sh
+sudo ls /etc/nginx/conf.d/mod-http-passenger.conf
+```
+
+## Configuring NGINX & Passenger
+
+
+Passenger comes with its own compiled Ruby version by default, I don't recommend using this as it can be outdated and we installed rbenv in advance which solves a lot of problems for us.
+
+To adjust this we need to edit the following file:
+
+```sh
+sudo nano /etc/nginx/conf.d/mod-http-passenger.conf
+```
+
+There we change the line to the following:
+
+```diff
+- passenger_ruby /usr/bin/passenger_free_ruby;
++ passenger_ruby /home/rails_user/.rbenv/shims/ruby;
```

Comment:
I checked now and found out that the folder `shims` is not existing in my home directory. Are we perhaps forgetting something here?

Reply:
If I install rbenv under the user as described this works. You can try `rbenv rehash` once and see if it works now.